### DISCUSSION TITLE
请问支持独立分包吗？ 我 import { defineComponent } from '@vue-mini/core'  报错

### DISCUSSION THREAD

**Discussion by houseCN** [vue-mini/vue-mini#116] [Q&A] [answered]
独立分包的页面，import { defineComponent } from '@vue-mini/core'，报错

是import路径没写对吗？

**Comment by yangmingshan** [comment, marked as answer]
这是个工程问题，不知道你的工程结构，可以发个 repo 看看

**Reply by houseCN** [reply]
1、npm create vue-mini@latest 创建项目后，在src目录下新增了 packageEntry/welcome/welcome

2、配置 app.json

3、运行项目，报错
[xcx-独立分包-报错-10.31.zip](https://github.com/user-attachments/files/17582165/xcx-.-.-10.31.zip)

![image](https://github.com/user-attachments/assets/5351e3a5-3b57-4deb-8994-d6ab7baa2ee7)

![image](https://github.com/user-attachments/assets/05928244-8df2-47c1-abda-45245511da08)

![image](https://github.com/user-attachments/assets/010518d0-94b7-49ab-a2d3-4aaf846e05f7)

**Reply by yangmingshan** [reply]
独立分包好像不支持 `miniprogram_npm` 啊，这就有点难搞了。

**Reply by houseCN** [reply]
OK，3Q
那就不用独立分包吧

**Reply by yangmingshan** [reply]
不对，是支持的。只是 `miniprogram_npm` 必须位于独立分包目录下。create-vue-mini 目前还不支持这种编译结构。

**Reply by yangmingshan** [reply]
后续可以跟踪这个 Issue：https://github.com/vue-mini/create-vue-mini/issues/26

**Reply by sonofmagic** [reply]
我在这个示例里面，做了一个独立分包的 demo

https://github.com/icebreaker-template/vue-mini-tailwindcss-template/tree/independent

**Reply by houseCN** [reply]
你这个用了vite，加了太多东西了呀

**Reply by sonofmagic** [reply]
> 你这个用了vite，加了太多东西了呀

分包拆包，独立构建其实是 weapp-vite 的能力，你也能够直接使用它进行这方面的处理。另外加的什么 tailwindcss , eslint 啥的，这些是我后面额外加的，用来提供其他一些强大的功能的。其实只要 weapp-vite 就够了。

**Reply by houseCN** [reply]
感谢
我的核心需求是使用vue-mini

**Reply by yangmingshan** [reply]
create-vue-mini v1.3.0 已支持独立分包

**Reply by houseCN** [reply]
> create-vue-mini v1.3.0 已支持独立分包

点赞，晚上回去就试试

**Reply by houseCN** [reply]
还是不行，报错如下：
![image](https://github.com/user-attachments/assets/c2c4df22-be49-4eed-964d-145724b842e2)

**Reply by yangmingshan** [reply]
这个错误应该是你使用的问题

**Reply by houseCN** [reply]
手动点击开发者工具“清缓存”和“编译”后，不报错了
可以了！
3Q
